### PR TITLE
rename persist-txn to txn-wal

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
This PR refactors existing code.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -131,6 +130,7 @@
 /src/testdrive                      @MaterializeInc/testing
 /src/timely-util                    @MaterializeInc/compute
 /src/transform                      @MaterializeInc/compute
+/src/txn-wal                        @aljoscha @danhhz @jkosh44
```
Comment (jkosh44) [inline .github/CODEOWNERS:133]: I wasn't really sure what to put here. Maybe it's time to make a Pv2 team in GH?

Comment (danhhz) [reply .github/CODEOWNERS:133]: +1 from me!

**Review by danhhz — APPROVED** [review]
Lgtm modulo the catalog changes and migration, I don't feel confident in my ability to reason about those, so it'd be lovely to get someone else to stamp that part

There were a couple name flips that I think probably could have stayed as they were before, but they were all marginal so I didn't bother nit-ing them

**Review by maddyblue — APPROVED** [review]

**Review by antiguru — APPROVED** [review]
Compute changes LGTM

**Pull request merged**